BrainPortal/cbrain_plugins/cbrain-plugins-base/cbrain_task/boutiques_task/views/_task_params.html.erb

```diff
@@ -34,6 +34,16 @@
   groups      = @descriptor.groups          || []
   params_in_groups = groups.map(&:members).flatten.uniq # list of param IDs
   interface_files  = Userfile.where(:id => @task.params[:interface_userfile_ids])
+
+  interface_files_name   = interface_files.pluck(:name)
```

Comment:
I don't understand this at all, and as a change to the framework I'm pretty it will crash in plenty of places. The partials expect interface_files to be an ActiveRecord relation and they will issue methods specific to AR (e.g. `.pluck()`) which are going to crash when they're given this array of small objects. Also, why the weird "name" value that contains a data provider name in parenthesis? If oyu wanted to chaneg the appearance of the names of the files in the form, you could replace in the view code `userfile.name` with the method `userfile.browse_name` instead, maybe?

Reply:
If you have an Array of Hash (it is my case here). 
You can use pluck, for example: 

> Userfile.last(10).map {|x| {type: x.type, name: x.name, id: x.id}}.pluck(:type) 

But right I can use browse_name instead.